tests/test_generics.py

```diff
@@ -33,6 +33,31 @@ class Result(GenericModel, Generic[data_type]):
     assert str(exc_info.value) == 'Cannot parameterize a concrete instantiation of a generic model'
 
 
+@skip_36
+def test_value_validation():
+    T = TypeVar('T')
+
+    class Response(GenericModel, Generic[T]):
+        data: T
+
+        @validator("data")
+        def validate_value_nonzero(cls, v):
+            if isinstance(v, dict):
+                return v  # ensure v is actually a value of the dict, not the dict itself
```

Comment:
another possible source of misbehavior (not currently broken, but since I wrote the test, figured I'd include it)